Add limits for issue comments

xmo-odoo opened the pull request:
When exceeding the allowable size, the error response claims that

> maximum is 65536 characters

but this seems to be off of emoji in the astral plane (which require 4 bytes in UTF-8):

- when posting pure ASCII, 262144 characters are allowed but 262145 are not
- when posting 🌈, 65536 occurrences (== 262144 bytes once encoded) are allowed but adding one ascii character to that (262145 bytes) is rejected.

Though I have not validated it I assume PR bodies have the same constraints so updated the qualifier there.

xmo-odoo committed:
Add limits for issue comments

When exceeding the allowable size, the error response claims that

> maximum is 65536 characters

but this seems to be off of emoji in the astral plane (which require 4 bytes in UTF-8):

- when posting pure ASCII, 262144 characters are allowed but 262145 are not
- when posting 🌈, 65536 occurrences (== 262144 bytes once encoded) are allowed but adding one ascii character to that (262145 bytes) is rejected.

Though I have not validated it I assume PR bodies have the same constraints so updated the qualifier there.

dead-claudia commented on README.md:
```diff
@@ -122,10 +122,16 @@ Note: length checking is performed after replacement
 
 - Min length: 0 characters
-- Max length: 262144 characters
+- Max length: 262144 bytes
```
As per your PR description, this was just assumed. Could you actually validate this?

xmo-odoo replied:
Hmmm after validation both the original limit and the updated version seem incorrect: I get a failure when I try to create a PR with a body of more than 65536 codepoints, whether those codepoints are ascii or astral. So for this one the error message that

> maximum is 65536 characters

is correct. This might have changed since the limit was originally tested.

(also I went back to check if the issue comment limit was still what I'd gotten last time around in case it'd changed since I previously checked it, I can confirm that the 262144 bytes limits is still correct as of 5 minutes ago).

xmo-odoo replied:
Checked both the PR and issue bodies, and they do match a limit of 65536 codepoints (I updated issues to the more precise term). I was able to confirm that this is specifically a codepoint limit by creating PRs with combining codepoints, both normalizable and not:

- "Latin Small Letter A with Caron" (U+01CE) repeated 65536 times works
- "Latin Small Letter A" (U+0061) followed by Combining Caron (U+030C), the decomposed form of the previous, fails
- "Latin Small Letter Y" (Y+0079) followed by Combining Caron (U+030C), which has no precomposed version, fails

dead-claudia commented on README.md:
```diff
@@ -122,10 +122,16 @@ Note: length checking is performed after replacement
 
+### Issue comments
+
+- min length: 1 character
```
```suggestion
- Min length: 1 character
```

dead-claudia commented on README.md:
```diff
@@ -122,10 +122,16 @@ Note: length checking is performed after replacement
+### Issue comments
+
+- min length: 1 character
+- Max length: 262144 bytes (65536~262144 characters depending on utf8-encoded size)
```
```suggestion
- Max length: 262144 bytes (65536-262144 characters depending on UTF8-encoded size)
```

xmo-odoo committed:
Casing / formatting fixes

Co-authored-by: Claudia Meadows <contact@claudiameadows.dev>

xmo-odoo committed:
Update limit for PR and issue bodies

Turns out the limit is 65536 codepoints, which matches what was specified for the issues.

dead-claudia commented on README.md:
```diff
@@ -114,18 +114,24 @@ Note: length checking is performed after replacement
+### Issue comments
+
+- Min length: 1 character
+- Max length: 262144 bytes (65536-262144 characters depending on UTF8-encoded size)
```
Did you forget to change this line as well per https://github.com/dead-claudia/github-limits/pull/13#discussion_r2509402890?

xmo-odoo replied:
Nope, frustratingly this one is correct as I wrote in https://github.com/dead-claudia/github-limits/pull/13#discussion_r2509342507 the size restrictions on issue/pr descriptions and on issue comments currently do not match.

dead-claudia committed:
Merge branch 'master' into xmo-odoo-patch-2
